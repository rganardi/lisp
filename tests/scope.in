(define 0
  (lambda (f)
    (lambda (x) x)))

(define 1
  (lambda (f)
    (lambda (x) (f x))))

(define +
  (lambda (m n)
    (lambda (f)
      (lambda (x)
	((m f) ((n f) x))))))

(define list
  (lambda x x))

;0
((0 (lambda x x)) (quote hi))
(((+ 0 0) (lambda x x)) (quote hi))
((0 list) (quote hi))

;1
((1 (lambda x x)) (quote hi))
((1 list) (quote hi))
(((+ 1 0) (lambda x x)) (quote hi))
(((+ 0 1) (lambda x x)) (quote hi))

;2
(((+ 1 1) list) (quote hi))
(((+ 1 1) (lambda x x)) (quote hi))
(((+ 0 (+ 1 1)) (lambda x x)) (quote hi))
(((+ (+ 1 1) 0) (lambda x x)) (quote hi))

;3
(((+ 1 (+ 1 1)) (lambda x x)) (quote hi))
(((+ (+ 1 1) 1) (lambda x x)) (quote hi))
(((+ (+ 1 1) 1) list) (quote hi))

;4
(((+ (+ (+ 1 1) 1) 1) (lambda x x)) (quote hi))
(((+ 1 (+ (+ 1 1) 1)) (lambda x x)) (quote hi))
