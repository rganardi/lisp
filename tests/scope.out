parse_eval: unspecified
parse_eval: unspecified
parse_eval: unspecified
parse_eval: unspecified
hi
hi
hi
(hi)
(hi)
(hi)
(hi)
((hi))
((hi))
((hi))
((hi))
(((hi)))
(((hi)))
(((hi)))
((((hi))))
((((hi))))
